main.cpp

```diff
@@ -19,13 +19,34 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter &rewriter) : rewriter_(rewriter) {};
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        const auto *castExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        const auto castIgnore = castExpr->getSubExprAsWritten()->IgnoreImpCasts();
+        auto &sourceManager = *Result.SourceManager;
+
+        auto type = Lexer::getSourceText(CharSourceRange::getTokenRange(

```

Comment:
Please descrbe the difference between the use of `getCharRange()` and `getTokenRange()`.

Reply:
1. `getCharRange()` returns the range from the first to the last character at the specified locations
2. `getTokenRange()` returns range of tokens before the start of the last token.

The function `getTokenRange()` will return **all tokens** in the specified range, while the function `getCharRange()` will return **only the string** contained in the specified range. 

For example, `getCharRange()` without specifying arguments, it can lead to the following error:
```
int main() {
    float variable;
    int i = static_cast<in>(variable); // skipped `t` in int type
    return 0;
}
```
So using `getTokenRange()` frees us from errors and indications of the correct length.